### PR TITLE
fix: entity element with named struct types not flattened

### PR DESCRIPTION
**Pull request by stockbal**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/).
 ### Added
 - Schema definition for `cds.typer` options in `package.json` and `.cdsrc-*.json` files
 
+### Fixed
+- Entity elements of named structured types are flatted when using the option `--inlineDeclarations flat`
```
Comment (daogrady) [inline CHANGELOG.md:12]:
```suggestion
- Entity elements of named structured types are flattened when using the option `--inlineDeclarations flat`
```

File: lib/visitor.js [hunk 2]
```diff
@@ -589,17 +593,19 @@ class Visitor {
      * @param {EntityCSN} element - CSN data belonging to the the element.
      * @param {SourceFile} file - the namespace file the surrounding entity is being printed into.
      * @param {Buffer} [buffer] - buffer to add the definition to. If no buffer is passed, the passed file's class buffer is used instead.
+     * @param {TypeResolveOptions} [resolverOptions] - custom type resolver options
      * @returns @see InlineDeclarationResolver.visitElement
      */
-    visitElement(name, element, file, buffer = file.classes) {
+    visitElement(name, element, file, buffer = file.classes, resolverOptions) {
```
Comment (daogrady) [inline lib/visitor.js:599]: This feels a bit awry, as it introduces an optional parameter without explicit default value after ones with explicit default values (sonarlint complains about this as well).
I think this should be streamlined with [`InlineDeclarationResolver::visitElement`](https://github.com/cap-js/cds-typer/blob/main/lib/components/inline.js#L78) where we pass a destructured object instead. That was due before your PR, but with the introduction of yet another parameter, this feels even more awkward. Could you please incorporate this change?

**Review by daogrady — CHANGES_REQUESTED** [review]
Thanks for this contribution! Looks mostly good, just one minor request wrt passing the newly introduced parameter and then we're good to go 🚀

File: lib/visitor.js (hunk 2 above)
Comment (stockbal) [reply lib/visitor.js]: Agreed. I made the adjustment

**Review by daogrady — APPROVED** [review]

**Pull request merged**